task4/apicode.py

```diff
@@ -0,0 +1,19 @@
+from flask import Flask, request, jsonify, session
+import redis
+
+app = Flask(__name__)
+cache = redis.StrictRedis(host='redis', port=6379, db=0)
+
+@app.route('/welcome', methods=["GET"])
+def Get():
+    name = request.args.get('name')
+    if(name):
```

Comment:
seems you are getting the name from query, from the description it should come from the database or cache https://github.com/Wiredcraft/test-devops#task-description-3

Reply:
I'm sorry for misunderstand that description, before that I think this GET method was running alone. I have submitted the updated file, please check again. Thanks.